app/components/Project/FormBase.tsx

```diff
@@ -0,0 +1,66 @@
+import FormBorder from "components/Layout/FormBorder";
+import { useEffect, useRef } from "react";
+import Form from "lib/theme/service-development-toolkit-form";
+import type { JSONSchema7 } from "json-schema";
+import FormComponentProps from "./FormComponentProps";
+
+interface Props extends FormComponentProps {
+  createInitialData: (data: object) => object;
+  schema: JSONSchema7;
+  uiSchema: object;
+}
+
+const FormBase: React.FunctionComponent<Props> = ({
+  formData,
+  applyChangesFromComponent,
+  onFormErrors,
+  createInitialData,
+  schema,
+  uiSchema,
+}) => {
+  // We restrict the data to only the fields we care about
+  // It's important to default to `undefined` so we trigger the errors on rjsf
+  const initialData = createInitialData(formData);
+
+  const makeErrorsObject = (errorSchema) => {
+    const keys = Object.keys(initialData);
+    const returnVal = {};
+    keys.forEach((key) => {
+      if (errorSchema[key]) {
+        returnVal[key] = errorSchema[key];
+      } else {
+        returnVal[key] = null;
+      }
+    });
+    return returnVal;
+  };
+
+  const formRef = useRef();
+
+  useEffect(() => {
+    const { errorSchema } = (formRef.current as any)?.validate(
+      initialData,
+      schema
+    );
+    onFormErrors(makeErrorsObject(errorSchema));
+  }, []);
+
+  return (
+    <FormBorder title="Background">
+      <Form
+        // @ts-ignore
+        ref={formRef}
+        schema={schema}
+        uiSchema={uiSchema}
+        formData={initialData}
+        onChange={(change) => {
+          applyChangesFromComponent(change.formData);
```

Comment:
does this callback mutate `change.formData`? From the name that's what it sounds like to me. If it's just a regular callback, perhaps this prop should be renamed to `onChange`?